torch_xla/csrc/tensor_methods.cpp

```diff
@@ -1921,7 +1921,9 @@ std::vector<XLATensor> XLATensor::split(const XLATensor& input,
   if (split_size == 0) {
     // Deal with 0 split size, it's a corner case which is only allowed when the
     // dimension size is 0 as well.
-    XLA_CHECK_EQ(dim_size, 0);
+    XLA_CHECK_EQ(dim_size, 0) << "split_size can only be 0 if dimension size "
```

Comment:
Hmm, I have doubts doing these kind of changes.

Reply:
why are the concerns? I'm adding it mainly because pytorch has test checking it, otherwise we'll have to skip certain tests when running pytorch test suite.

Reply:
So it checks for a specific error message?

Reply:
For this specific one it's not, https://github.com/pytorch/pytorch/blob/master/test/test_torch.py#L9096
But there're other checks with `assertRaisesRegex`. 
Looks like we do raise a RuntimeError now, it's possible that I created the issue just for a clearer error message. (I don't remember the reason exactly now) :P

Reply:
If we can avoid PR like this, it'd be better IMHO.

Reply:
Sure, IIUC the general guideline here is we only "fix" it when Pytorch test suite checks for `assertRaisesRegex`, and let it go if it only checks for `assertRaises`?

Reply:
In general, if there is a pytorch test failing because of that, it might be worth fixing it.
OTOH, I would not want to see 100 PRs like this to follow exact error messages :wink:

Reply:
sure, sgtm!